app/src/main/java/com/capstone/event_finder/activities/MainActivity.java

```diff
@@ -64,6 +67,50 @@ private void setUpBottomNavigation(FragmentManager fragmentManager) {
         bottomNavigationView.setSelectedItemId(R.id.action_feed);
     }
 
+    public void populateEventInfo(Event event, JsonObject temp) {
+        event.setName(temp.get("name").getAsString());
+        event.setDescription(temp.get("description").getAsString());
+        event.setImageUrl(temp.get("image_url").getAsString());
+        event.setTimeStart(temp.get("time_start").getAsString());
+        event.setId(temp.get("id").getAsString());
+        event.setEventSiteUrl(temp.get("event_site_url").getAsString());
+        event.setCategory(temp.get("category").getAsString());
+        checkAndSetEventEndTime(event, temp);
+        checkAndSetEventCost(event, temp);
+        formatAndSetEventLocation(temp, event);
+    }
+
+    private void checkAndSetEventCost(Event event, JsonObject temp) {
+        if (temp.get("cost").toString().matches("null")) {
+            event.setCost("N/A");
+        } else {
+            event.setCost("$" + temp.get("cost").getAsString() + "0");
+        }
+    }
+
+    private void checkAndSetEventEndTime(Event event, JsonObject temp) {
+        if (temp.get("time_end").toString().matches("null")) {
+            event.setTimeEnd(temp.get("time_start").getAsString());
+        } else {
+            event.setTimeEnd(temp.get("time_end").getAsString());
+        }
+    }
+
+    private void formatAndSetEventLocation(JsonObject temp, Event event) {
+        JsonArray formattedLocation = temp.getAsJsonObject("location").getAsJsonArray("display_address");
+        StringBuilder formattedLocationString = new StringBuilder();
+        for (int i = 0; i < formattedLocation.size(); i++) {
+            formattedLocationString.append(formattedLocation.get(i).getAsString()).append(" ");
+        }
+        event.setLocation(formattedLocationString.toString());
+    }
+
```

Comment:
Ignore the extra spacing here - realized I sent out the PR before deleting the space